### PR TITLE
baseline-604

### PR DESCRIPTION
**Pull request by shivang2311**
In the SearchSploit task, I focused on fixing a baseline bug by updating the component according to a predefined checklist. This included standardizing the code, removing unnecessary imports, and ensuring proper use of React hooks and components like LoadingOverlayAndCancelButtonPkexec. I also followed the project's style guide to ensure consistency.

### CODE REVIEW

**Review by Cyriacjose — CHANGES_REQUESTED** [review]
Hi @shivang2311 
The code still needs some work.

Please use Rendercomponent and dependencies to return.
Could you please kindly refer to AirbaseNG, or AircrackNG codes.

Regards

**Review by Cyriacjose — CHANGES_REQUESTED** [review]
Hi @shivang2311,

We are almost there just two changes required
Need Conditional Rendering ({!loadingModal && ...}): refer to Aircrack or Airbase code.
image

If you're not using the Switch component, you can remove its import statement from the first line.

**Comment by shivang2311** [comment]
made changes as told @Cyriacjose                                                                                                                                                                       Thanks!!

**Review by Cyriacjose — CHANGES_REQUESTED** [review]
Render Component not called

File: src/components/SearchSploit/SearchSploit.tsx [hunk 1]
```diff
@@ -1,137 +1,117 @@
...
+// Component Constants
+const title = "SearchSploit";
+const description = "SearchSploit is a command-line tool used for searching through Exploit-DB.";
+const steps = `Step 1: Enter a Search Term followed by selecting a Search Option.
+Step 2: Select an Output type.
+Step 3: Select a Non-Searching option.
+Step 4: Enter an Exploit Database ID.
+Step 5: Click Scan to commence SearchSploit's operation.
+Step 6: View the Output block below to view the results of the tool's execution.`;
+const sourceLink = "https://www.exploit-db.com/documentation/Offsec-SearchSploit.pdf";
```
Comment (Oliver2049) [inline src/components/SearchSploit/SearchSploit.tsx:29]: I think you used the wrong link for sourceLink, it should be like this:
```suggestion
const sourceLink = "https://www.kali.org/tools/exploitdb/#searchsploit";
```

File: src/components/SearchSploit/SearchSploit.tsx [hunk 2]
```diff
@@ -1,137 +1,117 @@
...
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
-import { Text } from "@mantine/core";
-import { List } from "@mantine/core";
-import { Accordion } from "@mantine/core";
-import { UserGuide } from "../UserGuide/UserGuide";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { LoadingOverlayAndCancelButtonPkexec } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
```
Comment (Oliver2049) [inline src/components/SearchSploit/SearchSploit.tsx:10]: You forgot  to call import { RenderComponent } from "../UserGuide/UserGuide";
```suggestion
import { RenderComponent } from "../UserGuide/UserGuide";
```

File: src/components/SearchSploit/SearchSploit.tsx [hunk 3]
```diff
@@ -142,122 +122,124 @@ const SearchSploit = () => {
...
-                <NativeSelect
-                    value={selectedNonSearchOption}
-                    onChange={(e) => setSelectedNonSearchOption(e.target.value)}
-                    label={"Non-Searching"}
-                    data={non_search}
-                    placeholder={"Select an option"}
-                    description={
-                        "Mirror (aka copies) an exploit to the current working directory, Examine (aka opens) the exploit using $PAGER"
-                    }
+        <>
```
Comment (Oliver2049) [inline src/components/SearchSploit/SearchSploit.tsx:176]: You for got to use RederComponent here
```suggestion
         <RenderComponent
            title={title}
            description={description}
            steps={steps}
            tutorial={tutorial}
            sourceLink={sourceLink}
        >
```

File: src/components/SearchSploit/SearchSploit.tsx [hunk 4]
```diff
@@ -142,122 +122,124 @@ const SearchSploit = () => {
...
+                                    ))}
+                                </List>
+                            </Accordion.Panel>
+                        </Accordion.Item>
+                    </Accordion>
+                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
+                    <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
+                </Stack>
+            </form>
+        </>
```
Comment (Oliver2049) [inline src/components/SearchSploit/SearchSploit.tsx:242]: Close RenderComponent here
```suggestion
        </RenderComponent>
```

File: src/components/SearchSploit/SearchSploit.tsx [hunk 5]
```diff
@@ -142,122 +122,124 @@ const SearchSploit = () => {
...
+                        label={"Non-Searching"}
+                        data={nonSearchOptions}
+                        placeholder={"Select an option"}
+                    />
+                    <TextInput
+                        label={"EBD-ID"}
+                        description="Exploit Database ID: Required when using the 'Path' output or Non-Search options."
+                        {...form.getInputProps("ebdId")}
+                    />
+                    <Button type={"submit"}>Scan</Button>
```
Comment (Oliver2049) [inline src/components/SearchSploit/SearchSploit.tsx:212]: You should use Start {title}
```suggestion
                    <Button type={"submit"}>Start {title}</Button>
```

**Review by Oliver2049 — APPROVED** [review]
Hi @shivang2311 
OVerall, this looks good to me. Thanks for your effort.
Regards, 
Oliver

**Review by Cyriacjose — APPROVED** [review]
Look good Tested on local environment seems working fine.
Great work @shivang2311

**Pull request merged**